File: cukcuk clone/app/src/main/java/vn/com/misa/cukcukstarterclone/data/repository/IUserRepository.java
package vn.com.misa.cukcukstarterclone.data.repository;

import vn.com.misa.cukcukstarterclone.data.IOnLoadedCallback;

/**
 * @created_by KhanhNQ on 08-Feb-2021.
 */
public interface IUserRepository {
    void login(String username, String password, IOnLoadedCallback<Boolean> callback);

    void register(String username, String password, IOnLoadedCallback<Boolean> callback);
}

File: app/src/main/java/com/misa/fresher/data/repository/IUserRepository.kt
package com.misa.fresher.data.repository

import com.misa.fresher.data.IOnLoadedCallback
import com.misa.fresher.data.model.User
import com.misa.fresher.data.model.UserError

/**
 * - interface's purpose:
 *
 * @author HTLong
 * @edit_at 2022-03-21
 */
interface IUserRepository {
    fun signIn(username: String, password: String, callback: IOnLoadedCallback<User, UserError>)
    fun signUp(username: String, password: String, callback: IOnLoadedCallback<User, UserError>)
}